appcues/src/main/java/com/appcues/Storage.kt

```diff
@@ -21,31 +25,85 @@ internal class Storage(
         context.getSharedPreferences("com.appcues.storage.${config.applicationId}", Context.MODE_PRIVATE)
 
     var deviceId: String
-        get() = sharedPreferences.getString(Constants.DeviceId.rawVal, null) ?: ""
-        set(value) = sharedPreferences.edit().putString(Constants.DeviceId.rawVal, value).apply()
+        get() = _deviceId
+        set(value) = updateDeviceId(value)
+
+    private lateinit var _deviceId: String
+
+    private fun updateDeviceId(deviceId: String) {
+        _deviceId = deviceId
+        appcuesCoroutineScope.launch {
+            sharedPreferences.edit().putString(Constants.DeviceId.rawVal, _deviceId).apply()
```

Comment:
from my earlier investigation, the `apply()` didn't actually need a coroutine launch, since its already doing an async save - correct?